src/Sylius/Component/Core/Promotion/Action/FixedDiscountPromotionActionCommand.php

```diff
@@ -61,20 +61,31 @@ public function execute(PromotionSubjectInterface $subject, array $configuration
             return false;
         }
 
-        $promotionAmount = $this->calculateAdjustmentAmount(
-            $subject->getPromotionSubjectTotal(),
-            $configuration[$channelCode]['amount']
-        );
+        $subjectTotal = $this->getSubjectTotal($subject, $promotion);
+        $promotionAmount = $this->calculateAdjustmentAmount($subjectTotal, $configuration[$channelCode]['amount']);
 
         if (0 === $promotionAmount) {
             return false;
         }
 
         if ($this->minimumPriceDistributor !== null) {
-            $splitPromotion = $this->minimumPriceDistributor->distribute($subject->getItems()->toArray(), $promotionAmount, $subject->getChannel());
+            $splitPromotion = $this->minimumPriceDistributor->distribute($subject->getItems()->toArray(), $promotionAmount, $subject->getChannel(), $promotion->getAppliesToDiscounted());
         } else {
             $itemsTotal = [];
             foreach ($subject->getItems() as $orderItem) {
+                if ($promotion->getAppliesToDiscounted()) {
+                    $itemsTotal[] = $orderItem->getTotal();
+
+                    continue;
+                }
+
+                $variant = $orderItem->getVariant();
+                if (!empty($variant->getAppliedPromotionsForChannel($subject->getChannel()))) {
+                    $itemsTotal[] = 0;
+
+                    continue;
+                }
+
                 $itemsTotal[] = $orderItem->getTotal();
             }
```

Comment:
My soul hurts, as this part of code is identical to [this one](https://github.com/Sylius/Sylius/pull/13349/files#diff-a1b12f5c3dada5f3a732a1789bcaafd6876f7a4d33088670dff3126ad28e3219R67-R89) 😢